### DISCUSSION TITLE
libssh2_userauth_publickey_fromfile returning -19, "Callback returned error"

### DISCUSSION THREAD

**Discussion by aguaviva** [libssh2/libssh2#1255] [Q&A]
How can I tell what is the reason for this error? I am using 
- libgcrypt-1-10-3
- a ecdsa key in openSSH format

I have seen similar discussions and read the docs but nothing really addresses my issue.

Thanks

**Comment by aguaviva** [comment]
I just read the key has to be encoded in PEM format

**Reply by apteryks** [reply]
So this is resolved?  If so, could you mark it as answered?
